weave/Scripts/CurveSpawner.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using Godot;
+using GodotSharper.Instancing;
+using weave;
+
+[Instantiable("res://Objects/CurveSpawner.tscn")]
+public partial class CurveSpawner : Node2D
+{
+    public Player Player { get; set; }
+    public List<SegmentShape2D> Segments { get; set; } = new();
+    public int LineWidth { get; set; }
+    private readonly float curveSpawnOffset = 4f; 
+    private Vector2 lastPoint;
+    private bool hasStarted = false;
+    private Color lineColor = new(1, 0, 0);
```

Comment:
Multiple of these fields dont follow C# naming conventions, will try to configure Roslynator CLI to "complain" about these